models/classes/Copier/InstanceContentCopier.php

```diff
@@ -0,0 +1,103 @@
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ *
+ * @author Andrei Shapiro <andrei.shapiro@taotesting.com>
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoItems\model\Copier;
+
+use core_kernel_classes_Resource;
+use oat\oatbox\event\EventManager;
+use oat\oatbox\filesystem\Directory;
+use taoItems_models_classes_ItemsService;
+use oat\taoItems\model\event\ItemContentClonedEvent;
+use oat\generis\model\fileReference\FileReferenceSerializer;
+use oat\tao\model\resources\Contract\InstanceContentCopierInterface;
+
+class InstanceContentCopier implements InstanceContentCopierInterface
+{
+    public const PROPERTY_ITEM_CONTENT = 'http://www.tao.lu/Ontologies/TAOItem.rdf#ItemContent';
+    private const PROPERTY_ITEM_MODEL = 'http://www.tao.lu/Ontologies/TAOItem.rdf#ItemModel';
+
+    /** @var FileReferenceSerializer */
+    private $fileReferenceSerializer;
+
+    /** @var taoItems_models_classes_ItemsService */
+    private $itemsService;
+
+    /** @var EventManager */
+    private $eventManager;
+
+    public function __construct(
+        FileReferenceSerializer $fileReferenceSerializer,
+        taoItems_models_classes_ItemsService $itemsService,
+        EventManager $eventManager
+    ) {
+        $this->fileReferenceSerializer = $fileReferenceSerializer;
+        $this->itemsService = $itemsService;
+        $this->eventManager = $eventManager;
+    }
+
+    public function copy(
+        core_kernel_classes_Resource $instance,
+        core_kernel_classes_Resource $destinationInstance
+    ): void {
+        $property = $instance->getProperty(self::PROPERTY_ITEM_CONTENT);
+        $itemModelProperty = $instance->getProperty(self::PROPERTY_ITEM_MODEL);
+
+        $model = $instance->getOnePropertyValue($itemModelProperty);
+        $model = $model instanceof core_kernel_classes_Resource ? $model : null;
+        $destinationInstance->editPropertyValues($itemModelProperty, $model);
+
+        foreach ($instance->getUsedLanguages($property) as $lang) {
+            $sourceItemDirectory = $this->itemsService->getItemDirectory($instance, $lang);
+            $destinationItemDirectory = $this->itemsService->getItemDirectory($destinationInstance, $lang);
+            $propertyValues = $instance
+                ->getPropertyValuesCollection($property, ['lg' => $lang])
+                ->getIterator();
+
+            foreach ($propertyValues as $propertyValue) {
```

Comment:
suggestion: It may be slightly more clear to just get the values collection first and then inline the `getIterator()` call.

```suggestion
            $propertyValues = $instance
                ->getPropertyValuesCollection($property, ['lg' => $lang]);

            foreach ($propertyValues->getIterator() as $propertyValue) {
```

Reply:
Also, `core_kernel_classes_ContainerCollection` implements `IteratorAggregate`, so we may also omit the `getIterator()` call altogether:

```suggestion
            $propertyValues = $instance->getPropertyValuesCollection($property, ['lg' => $lang]);

            foreach ($propertyValues as $propertyValue) {
```